flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -4122,13 +4197,16 @@ class ArrayExprLowering {
   Fortran::lower::SymMap &symMap;
   llvm::Optional<CC> ccDest;
   fir::ArrayLoadOp destination;
+  const fir::MutableBoxValue *destinationMutableBox{};
```

Comment:
Maybe a comment here? I think there is a constraint that `destination` is set or `destinationMutableBox` or neither, oui?